<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>HPAclusteR ‚Ä¢ HPAclusteR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="‚Äùimage/svg+xml‚Äù" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="HPAclusteR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HPAclusteR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/HPAclusteR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/buenoalvezm/HPAclusteR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>HPAclusteR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/buenoalvezm/HPAclusteR/blob/master/vignettes/HPAclusteR.Rmd" class="external-link"><code>vignettes/HPAclusteR.Rmd</code></a></small>
      <div class="d-none name"><code>HPAclusteR.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>HPAclusteR</code> is an R package designed by the Human Protein
Atlas to streamline the process of gene clustering from transcriptomics
data. It provides a modular pipeline for performing PCA, calculating
distances, constructing shared nearest neighbor (SNN) graphs, consensus
clustering, UMAP, functional annotation, and different publication-ready
visualizations. The package is designed to work with <a href="https://github.com/emiliosk/AnnDatR" class="external-link">AnnDatR objects</a>, making
it easy to integrate into workflows for transcriptomics data
analysis.</p>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>To begin, load the <code>HPAclusteR</code> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://buenoalvezm.github.io/HPAclusteR/">HPAclusteR</a></span><span class="op">)</span></span></code></pre></div>
<p>We will use the built-in <code>example_adata</code> dataset for this
vignette. This dataset is a toy example designed to demonstrate the
functionality of the package.</p>
<blockquote>
<p>üí° HPAclusteR can work with AnnDatR objects created using the AnnDatR
package, as long as the data is transposed so that genes are the
observations (rows) and samples are the variables (columns).
Alternatively, you can use the hc_initialize() function to create a
properly formatted object directly from your data frames.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="clustering-pipeline">Clustering Pipeline<a class="anchor" aria-label="anchor" href="#clustering-pipeline"></a>
</h2>
<blockquote>
<p>üí° This part demonstrates the step-by-step usage of the
<code>HPAclusteR</code> pipeline. For users who prefer simplicity, the
<code><a href="../reference/hc_auto_cluster.html">hc_auto_cluster()</a></code> function can be used to run the entire
pipeline in one call.</p>
</blockquote>
<div class="section level3">
<h3 id="step-1-principal-component-analysis-pca">Step 1: Principal Component Analysis (PCA)<a class="anchor" aria-label="anchor" href="#step-1-principal-component-analysis-pca"></a>
</h3>
<p>PCA is the first step in the pipeline, used to reduce the
dimensionality of the data while retaining the most important
features.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adata_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_pca.html">hc_pca</a></span><span class="op">(</span><span class="va">example_adata</span>, components <span class="op">=</span> <span class="fl">40</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">adata_res</span><span class="op">$</span><span class="va">obsm</span><span class="op">$</span><span class="va">X_pca</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span>  <span class="co"># PCA scores</span></span>
<span><span class="co">#&gt;                        PC1       PC2        PC3        PC4        PC5</span></span>
<span><span class="co">#&gt; ENSG00000002745  2.5250898 -1.245267  4.9538170 -2.9612872  0.5522311</span></span>
<span><span class="co">#&gt; ENSG00000004660 -3.9791370 -4.237506  1.0128910 -2.3799084  0.1714757</span></span>
<span><span class="co">#&gt; ENSG00000006047 -3.3357725  5.230900  0.2779654  0.8539579 -1.0689779</span></span>
<span><span class="co">#&gt; ENSG00000006059  0.3849729  1.662731  4.2587010 -1.7739603  0.3751746</span></span>
<span><span class="co">#&gt; ENSG00000006453  4.3272116  2.237562  2.0914290  0.3764680 -5.5500274</span></span>
<span><span class="co">#&gt; ENSG00000006740 -3.4995078 -4.826965 -1.0965416  0.6870355 -3.5294865</span></span>
<span><span class="va">adata_res</span><span class="op">$</span><span class="va">uns</span><span class="op">$</span><span class="va">pca</span>  <span class="co"># Explained variance</span></span>
<span><span class="co">#&gt; nipals calculated PCA</span></span>
<span><span class="co">#&gt; Importance of component(s):</span></span>
<span><span class="co">#&gt;                  PC1    PC2     PC3     PC4     PC5     PC6     PC7     PC8</span></span>
<span><span class="co">#&gt; R2            0.2353 0.1895 0.09884 0.07956 0.06474 0.05336 0.02484 0.02105</span></span>
<span><span class="co">#&gt; Cumulative R2 0.2353 0.4248 0.52366 0.60323 0.66797 0.72133 0.74617 0.76721</span></span>
<span><span class="co">#&gt;                  PC9    PC10    PC11    PC12    PC13     PC14     PC15     PC16</span></span>
<span><span class="co">#&gt; R2            0.0169 0.01463 0.01307 0.01153 0.01092 0.009315 0.008165 0.007873</span></span>
<span><span class="co">#&gt; Cumulative R2 0.7841 0.79874 0.81180 0.82333 0.83426 0.843573 0.851738 0.859611</span></span>
<span><span class="co">#&gt;                   PC17     PC18    PC19    PC20     PC21     PC22     PC23</span></span>
<span><span class="co">#&gt; R2            0.007254 0.007048 0.00658 0.00588 0.005541 0.005381 0.005207</span></span>
<span><span class="co">#&gt; Cumulative R2 0.866864 0.873913 0.88049 0.88637 0.891914 0.897295 0.902502</span></span>
<span><span class="co">#&gt;                   PC24     PC25     PC26     PC27     PC28     PC29     PC30</span></span>
<span><span class="co">#&gt; R2            0.004988 0.004932 0.004718 0.004567 0.004482 0.004177 0.003912</span></span>
<span><span class="co">#&gt; Cumulative R2 0.907490 0.912421 0.917139 0.921706 0.926188 0.930364 0.934277</span></span>
<span><span class="co">#&gt;                   PC31     PC32     PC33     PC34     PC35     PC36     PC37</span></span>
<span><span class="co">#&gt; R2            0.003712 0.003605 0.003258 0.003229 0.003022 0.002936 0.002783</span></span>
<span><span class="co">#&gt; Cumulative R2 0.937988 0.941593 0.944851 0.948080 0.951102 0.954038 0.956821</span></span>
<span><span class="co">#&gt;                   PC38     PC39     PC40</span></span>
<span><span class="co">#&gt; R2            0.002728 0.002528 0.002346</span></span>
<span><span class="co">#&gt; Cumulative R2 0.959549 0.962078 0.964423</span></span>
<span><span class="co">#&gt; 78   Variables</span></span>
<span><span class="co">#&gt; 981  Samples</span></span>
<span><span class="co">#&gt; 111  NAs ( 0.145 %)</span></span>
<span><span class="co">#&gt; 40   Calculated component(s)</span></span>
<span><span class="co">#&gt; Data was mean centered before running PCA </span></span>
<span><span class="co">#&gt; Data was NOT scaled before running PCA </span></span>
<span><span class="co">#&gt; Scores structure:</span></span>
<span><span class="co">#&gt; [1] 981  40</span></span>
<span><span class="co">#&gt; Loadings structure:</span></span>
<span><span class="co">#&gt; [1] 78 40</span></span></code></pre></div>
<p>The PCA step reduces the dimensionality of the data to 40 components,
which can be adjusted based on the dataset and analysis goals.</p>
</div>
<div class="section level3">
<h3 id="step-2-distance-calculation">Step 2: Distance Calculation<a class="anchor" aria-label="anchor" href="#step-2-distance-calculation"></a>
</h3>
<p>Next, we calculate the pairwise distances between samples using the
PCA-reduced data. The <code><a href="../reference/hc_distance.html">hc_distance()</a></code> function supports
multiple distance metrics, such as ‚Äúeuclidean‚Äù and ‚Äúspearman‚Äù. We will
use the number of components determined by Kaiser‚Äôs rule.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adata_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_distance.html">hc_distance</a></span><span class="op">(</span></span>
<span>  <span class="va">adata_res</span>, </span>
<span>  components <span class="op">=</span> <span class="fu"><a href="../reference/hc_kaisers_rule.html">hc_kaisers_rule</a></span><span class="op">(</span><span class="va">adata_res</span><span class="op">)</span>, </span>
<span>  method <span class="op">=</span> <span class="st">"spearman"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Kaiser's rule is above 80% variation. Success"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">adata_res</span><span class="op">$</span><span class="va">uns</span><span class="op">$</span><span class="va">distance</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.9545455 1.3545455 0.4727273 0.8636364 1.1818182 1.3909091</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-3-shared-nearest-neighbor-snn-graph-construction">Step 3: Shared Nearest Neighbor (SNN) Graph Construction<a class="anchor" aria-label="anchor" href="#step-3-shared-nearest-neighbor-snn-graph-construction"></a>
</h3>
<p>The SNN graph is constructed to identify clusters of similar samples.
This step uses the distance matrix calculated in the previous step.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adata_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_snn.html">hc_snn</a></span><span class="op">(</span><span class="va">adata_res</span>, neighbors <span class="op">=</span> <span class="fl">15</span>, prune <span class="op">=</span> <span class="fl">1</span> <span class="op">/</span> <span class="fl">15</span><span class="op">)</span></span>
<span><span class="va">adata_res</span><span class="op">$</span><span class="va">uns</span><span class="op">$</span><span class="va">neighbors</span><span class="op">$</span><span class="va">snn</span></span>
<span><span class="co">#&gt; A Graph object containing 981 cells</span></span></code></pre></div>
<p>The <code>neighbors</code> parameter controls the number of nearest
neighbors considered, while <code>prune</code> adjusts the sparsity of
the graph.</p>
</div>
<div class="section level3">
<h3 id="step-4-consensus-clustering">Step 4: Consensus Clustering<a class="anchor" aria-label="anchor" href="#step-4-consensus-clustering"></a>
</h3>
<p>Consensus clustering is performed to identify robust clusters in the
data. This step aggregates clustering results from multiple runs, in
this case 100.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adata_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_cluster_consensus.html">hc_cluster_consensus</a></span><span class="op">(</span></span>
<span>  <span class="va">adata_res</span>, </span>
<span>  resolution <span class="op">=</span> <span class="fl">8</span>, </span>
<span>  method <span class="op">=</span> <span class="st">"louvain"</span>,</span>
<span>  n_seeds <span class="op">=</span> <span class="fl">100</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">adata_res</span><span class="op">$</span><span class="va">obs</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "30" "14" "4"  "11" "21" "14"</span></span></code></pre></div>
<p>The <code>resolution</code> parameter controls the granularity of the
clustering. Higher values result in more clusters.</p>
</div>
<div class="section level3">
<h3 id="step-5-umap-visualization">Step 5: UMAP Visualization<a class="anchor" aria-label="anchor" href="#step-5-umap-visualization"></a>
</h3>
<p>UMAP is used to visualize the clusters in a low-dimensional space.
This step provides an intuitive way to explore the clustering results.
It uses the constructed SNN graph.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adata_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_umap.html">hc_umap</a></span><span class="op">(</span><span class="va">adata_res</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">adata_res</span><span class="op">$</span><span class="va">obsm</span><span class="op">$</span><span class="va">X_umap</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     UMAP_1     UMAP_2</span></span>
<span><span class="co">#&gt; ENSG00000002745  0.2972319  0.5904026</span></span>
<span><span class="co">#&gt; ENSG00000004660 -0.9859446 -0.4686754</span></span>
<span><span class="co">#&gt; ENSG00000006047  0.9013946 -0.4792866</span></span>
<span><span class="co">#&gt; ENSG00000006059  0.1685648  0.4550808</span></span>
<span><span class="co">#&gt; ENSG00000006453 -0.4699026 -0.3761350</span></span>
<span><span class="co">#&gt; ENSG00000006740 -0.9678060 -0.4816415</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-6-cluster-hulls-optional">Step 6: Cluster Hulls (Optional)<a class="anchor" aria-label="anchor" href="#step-6-cluster-hulls-optional"></a>
</h3>
<p>Cluster hulls are calculated to visualize the boundaries of each
cluster in the UMAP plot.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adata_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_cluster_hulls.html">hc_cluster_hulls</a></span><span class="op">(</span><span class="va">adata_res</span>, poly_smoothing <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">adata_res</span><span class="op">$</span><span class="va">uns</span><span class="op">$</span><span class="va">UMAP_hulls</span><span class="op">$</span><span class="va">hulls</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 √ó 7</span></span></span>
<span><span class="co">#&gt;   cluster sub_cluster sub_type landmass      X      Y polygon_id</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 14                1 primary         1 -<span style="color: #BB0000;">0.980</span> -<span style="color: #BB0000;">0.502</span> 14_1_1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 14                1 primary         1 -<span style="color: #BB0000;">1.00</span>  -<span style="color: #BB0000;">0.502</span> 14_1_1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 14                1 primary         1 -<span style="color: #BB0000;">1.00</span>  -<span style="color: #BB0000;">0.474</span> 14_1_1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 14                1 primary         1 -<span style="color: #BB0000;">0.987</span> -<span style="color: #BB0000;">0.46</span>  14_1_1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 14                1 primary         1 -<span style="color: #BB0000;">0.959</span> -<span style="color: #BB0000;">0.46</span>  14_1_1    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 14                1 primary         1 -<span style="color: #BB0000;">0.952</span> -<span style="color: #BB0000;">0.495</span> 14_1_1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-7-visualization">Step 7: Visualization<a class="anchor" aria-label="anchor" href="#step-7-visualization"></a>
</h3>
<p>Finally, we visualize the UMAP plot with clusters and hulls using the
<code><a href="../reference/hc_plot_umap.html">hc_plot_umap()</a></code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hc_plot_umap.html">hc_plot_umap</a></span><span class="op">(</span><span class="va">adata_res</span>, plot <span class="op">=</span> <span class="st">"both"</span><span class="op">)</span></span></code></pre></div>
<p><img src="HPAclusteR_files/figure-html/plot-1.png" class="r-plt" alt="" width="700"></p>
<p>The <code>plot</code> argument can be set to <code>"points"</code>,
<code>"hulls"</code>, or <code>"both"</code> to customize the
visualization.</p>
</div>
</div>
<div class="section level2">
<h2 id="cluster-annotation">Cluster Annotation<a class="anchor" aria-label="anchor" href="#cluster-annotation"></a>
</h2>
<p>In this part we will annotate our clusters using the KEGG database.
Users can use KEGG, GO or even other databases such as the Human Protein
Atlas, Panglao, Trrust and Reactome. Check <code><a href="../reference/hc_annotate.html">hc_annotate()</a></code>
for further details.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enrichment_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_annotate.html">hc_annotate</a></span><span class="op">(</span></span>
<span>  <span class="va">adata_res</span>,</span>
<span>  dbs <span class="op">=</span> <span class="st">"KEGG"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in clusterProfiler::bitr(unique(clustering_data[["gene"]]), fromType =</span></span>
<span><span class="co">#&gt; "ENSEMBL", : 0.82% of input gene IDs are fail to map...</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">enrichment_res</span><span class="op">$</span><span class="va">enrichment</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 √ó 10</span></span></span>
<span><span class="co">#&gt;   `Cluster ID` Database      `Term ID` Term          GeneRatio BgRatio `P-value`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 11           KEGG pathways hsa04382  Cornified en‚Ä¶ 13/24     40/350    8.38<span style="color: #949494;">e</span><span style="color: #BB0000;">-8</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 11           KEGG pathways hsa04915  Estrogen sig‚Ä¶ 4/24      12/350    5.86<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 12           KEGG pathways hsa01100  Metabolic pa‚Ä¶ 9/15      67/350    4.03<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 13           KEGG pathways hsa04814  Motor protei‚Ä¶ 3/11      14/350    6.99<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 17           KEGG pathways hsa03040  Spliceosome   5/20      13/350    3.49<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 2            KEGG pathways hsa03082  ATP-dependen‚Ä¶ 3/5       10/350    1.64<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ‚Ñπ 3 more variables: `Adjusted P-value` &lt;dbl&gt;, `Gene IDs` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Gene names` &lt;chr&gt;</span></span></span>
<span><span class="va">enrichment_res</span><span class="op">$</span><span class="va">bubblemap_kegg</span></span></code></pre></div>
<p><img src="HPAclusteR_files/figure-html/annotate-1.png" class="r-plt" alt="" width="700"></p>
<p>We can also perform gene classification within each cluster using the
Human Protein Atlas (HPA) logic. This approach assigns specificity and
distribution categories to genes based on their expression patterns
across sample categories (e.g., tissues), providing insights into gene
function and tissue specificity.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">classify_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_classify.html">hc_classify</a></span><span class="op">(</span></span>
<span>  <span class="va">adata_res</span>,</span>
<span>  sample_categories <span class="op">=</span> <span class="st">"tissue_name"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">classify_res</span><span class="op">$</span><span class="va">classification</span><span class="op">[[</span><span class="st">"1"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 √ó 5</span></span></span>
<span><span class="co">#&gt;   ENSG            spec_category   spec_sample_categories   tau dist_category   </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ENSG00000069974 Enhanced        stomach;bone marrow     0.43 Detected in all </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ENSG00000072501 Low specificity <span style="color: #BB0000;">NA</span>                      0.35 Detected in all </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ENSG00000077147 Low specificity <span style="color: #BB0000;">NA</span>                      0.28 Detected in all </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ENSG00000078177 Enhanced        thymus                  0.64 Detected in many</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> ENSG00000081923 Enhanced        colon;rectum;stomach    0.59 Detected in many</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> ENSG00000101868 Enhanced        thymus                  0.51 Detected in many</span></span>
<span><span class="va">classify_res</span><span class="op">$</span><span class="va">pie_charts</span><span class="op">[[</span><span class="st">"1"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="HPAclusteR_files/figure-html/classify-1.png" class="r-plt" alt="" width="700"></p>
<p>We can visualize the gene expression patterns within each cluster
using heatmaps. Both z-score and scaled expression heatmaps are
generated for each cluster, allowing us to explore and compare gene
expression profiles across different groups. The confidence of each
cluster is also presented in the sidebar.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_heatmaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_plot_expression.html">hc_plot_expression</a></span><span class="op">(</span><span class="va">adata_res</span><span class="op">)</span></span>
<span><span class="va">expr_heatmaps</span><span class="op">$</span><span class="va">zscore</span><span class="op">[[</span><span class="st">"1"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="HPAclusteR_files/figure-html/expression-1.png" class="r-plt" alt="" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr_heatmaps</span><span class="op">$</span><span class="va">scaled</span><span class="op">[[</span><span class="st">"1"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="HPAclusteR_files/figure-html/expression-2.png" class="r-plt" alt="" width="700"></p>
</div>
<div class="section level2">
<h2 id="cluster-comparison">Cluster Comparison<a class="anchor" aria-label="anchor" href="#cluster-comparison"></a>
</h2>
<p>Finally, we can compare different clustering results using the
<code><a href="../reference/hc_cluster_compare.html">hc_cluster_compare()</a></code> function. Here, we will create a
second clustering result with a different resolution and compare it to
the first result. This time we will use the
<code><a href="../reference/hc_auto_cluster.html">hc_auto_cluster()</a></code> function for simplicity.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adata_res2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_auto_cluster.html">hc_auto_cluster</a></span><span class="op">(</span></span>
<span>  <span class="va">example_adata</span>,</span>
<span>  cluster_resolution <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Kaiser's rule is above 80% variation. Success"</span></span></code></pre></div>
<p><img src="HPAclusteR_files/figure-html/comparison-1.png" class="r-plt" alt="" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">comparison_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hc_cluster_compare.html">hc_cluster_compare</a></span><span class="op">(</span></span>
<span>  <span class="va">adata_res</span>,</span>
<span>  <span class="va">adata_res2</span>,</span>
<span>  graph_type <span class="op">=</span> <span class="st">"bipartite"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">‚Ñπ</span> Please use the `linewidth` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">‚Ñπ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">HPAclusteR</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at</span></span>
<span><span class="co">#&gt;   <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/buenoalvezm/HPAclusteR/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once per session.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span>
<span><span class="va">comparison_res</span><span class="op">$</span><span class="va">heatmap</span></span></code></pre></div>
<p><img src="HPAclusteR_files/figure-html/comparison-2.png" class="r-plt" alt="" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparison_res</span><span class="op">$</span><span class="va">network</span></span></code></pre></div>
<p><img src="HPAclusteR_files/figure-html/comparison-3.png" class="r-plt" alt="" width="700"></p>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>Explore the package further to uncover the full potential of your
transcriptomics data!</p>
<blockquote>
<p>üí° Remember that these data are an example toy-dataset with only a
sample of the total protein-coding genes. The results in this guide
should not be interpreted as real results. The purpose of this vignette
is to show you how to use the package and its functions.</p>
</blockquote>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] HPAclusteR_1.0.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] fs_1.6.6                matrixStats_1.5.0       spatstat.sparse_3.1-0  </span></span>
<span><span class="co">#&gt;   [4] enrichplot_1.30.4       httr_1.4.7              RColorBrewer_1.1-3     </span></span>
<span><span class="co">#&gt;   [7] prabclus_2.3-5          tools_4.5.2             sctransform_0.4.3      </span></span>
<span><span class="co">#&gt;  [10] utf8_1.2.6              R6_2.6.1                lazyeval_0.2.2         </span></span>
<span><span class="co">#&gt;  [13] uwot_0.2.4              withr_3.0.2             sp_2.2-0               </span></span>
<span><span class="co">#&gt;  [16] GGally_2.4.0            gridExtra_2.3           progressr_0.18.0       </span></span>
<span><span class="co">#&gt;  [19] cli_3.6.5               Biobase_2.70.0          textshaping_1.0.4      </span></span>
<span><span class="co">#&gt;  [22] factoextra_1.0.7        spatstat.explore_3.7-0  fastDummies_1.7.5      </span></span>
<span><span class="co">#&gt;  [25] network_1.20.0          scatterpie_0.2.6        slam_0.1-55            </span></span>
<span><span class="co">#&gt;  [28] labeling_0.4.3          sass_0.4.10             diptest_0.77-2         </span></span>
<span><span class="co">#&gt;  [31] Seurat_5.4.0            tm_0.7-17               S7_0.2.1               </span></span>
<span><span class="co">#&gt;  [34] robustbase_0.99-7       spatstat.data_3.1-9     readr_2.1.6            </span></span>
<span><span class="co">#&gt;  [37] askpass_1.2.1           ggridges_0.5.7          pbapply_1.7-4          </span></span>
<span><span class="co">#&gt;  [40] pkgdown_2.2.0           systemfonts_1.3.1       yulab.utils_0.2.4      </span></span>
<span><span class="co">#&gt;  [43] gson_0.1.0              DOSE_4.4.0              R.utils_2.13.0         </span></span>
<span><span class="co">#&gt;  [46] parallelly_1.46.1       RSQLite_2.4.6           treemap_2.4-4          </span></span>
<span><span class="co">#&gt;  [49] generics_0.1.4          gridGraphics_0.5-1      ica_1.0-3              </span></span>
<span><span class="co">#&gt;  [52] spatstat.random_3.4-4   dplyr_1.2.0             GO.db_3.22.0           </span></span>
<span><span class="co">#&gt;  [55] Matrix_1.7-4            S4Vectors_0.48.0        abind_1.4-8            </span></span>
<span><span class="co">#&gt;  [58] R.methodsS3_1.8.2       lifecycle_1.0.5         yaml_2.3.12            </span></span>
<span><span class="co">#&gt;  [61] qvalue_2.42.0           Rtsne_0.17              grid_4.5.2             </span></span>
<span><span class="co">#&gt;  [64] blob_1.3.0              promises_1.5.0          crayon_1.5.3           </span></span>
<span><span class="co">#&gt;  [67] miniUI_0.1.2            ggtangle_0.1.1          lattice_0.22-7         </span></span>
<span><span class="co">#&gt;  [70] cowplot_1.2.0           KEGGREST_1.50.0         sna_2.8                </span></span>
<span><span class="co">#&gt;  [73] pillar_1.11.1           knitr_1.51              fgsea_1.36.2           </span></span>
<span><span class="co">#&gt;  [76] fpc_2.2-14              future.apply_1.20.1     codetools_0.2-20       </span></span>
<span><span class="co">#&gt;  [79] fastmatch_1.1-8         glue_1.8.0              ggiraph_0.9.4          </span></span>
<span><span class="co">#&gt;  [82] V8_8.0.1                ggfun_0.2.0             spatstat.univar_3.1-6  </span></span>
<span><span class="co">#&gt;  [85] fontLiberation_0.1.0    pcaMethods_2.2.0        data.table_1.18.2.1    </span></span>
<span><span class="co">#&gt;  [88] vctrs_0.7.1             png_0.1-8               treeio_1.34.0          </span></span>
<span><span class="co">#&gt;  [91] spam_2.11-3             gtable_0.3.6            kernlab_0.9-33         </span></span>
<span><span class="co">#&gt;  [94] cachem_1.1.0            xfun_0.56               mime_0.13              </span></span>
<span><span class="co">#&gt;  [97] Seqinfo_1.0.0           coda_0.19-4.1           survival_3.8-3         </span></span>
<span><span class="co">#&gt; [100] pheatmap_1.0.13         fitdistrplus_1.2-6      ROCR_1.0-12            </span></span>
<span><span class="co">#&gt; [103] nlme_3.1-168            ggtree_4.0.4            bit64_4.6.0-1          </span></span>
<span><span class="co">#&gt; [106] fontquiver_0.2.1        RcppAnnoy_0.0.23        bslib_0.10.0           </span></span>
<span><span class="co">#&gt; [109] irlba_2.3.7             KernSmooth_2.23-26      otel_0.2.0             </span></span>
<span><span class="co">#&gt; [112] colorspace_2.1-2        rrvgo_1.22.0            BiocGenerics_0.56.0    </span></span>
<span><span class="co">#&gt; [115] DBI_1.2.3               nnet_7.3-20             tidyselect_1.2.1       </span></span>
<span><span class="co">#&gt; [118] bit_4.6.0               compiler_4.5.2          curl_7.0.0             </span></span>
<span><span class="co">#&gt; [121] xml2_1.5.2              NLP_0.3-2               desc_1.4.3             </span></span>
<span><span class="co">#&gt; [124] fontBitstreamVera_0.1.1 plotly_4.12.0           scales_1.4.0           </span></span>
<span><span class="co">#&gt; [127] DEoptimR_1.1-4          lmtest_0.9-40           rappdirs_0.3.4         </span></span>
<span><span class="co">#&gt; [130] stringr_1.6.0           digest_0.6.39           goftest_1.2-3          </span></span>
<span><span class="co">#&gt; [133] spatstat.utils_3.2-1    rmarkdown_2.30          XVector_0.50.0         </span></span>
<span><span class="co">#&gt; [136] htmltools_0.5.9         pkgconfig_2.0.3         umap_0.2.10.0          </span></span>
<span><span class="co">#&gt; [139] fastmap_1.2.0           rlang_1.1.7             htmlwidgets_1.6.4      </span></span>
<span><span class="co">#&gt; [142] shiny_1.12.1            farver_2.1.2            jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [145] zoo_1.8-15              jsonlite_2.0.0          statnet.common_4.13.0  </span></span>
<span><span class="co">#&gt; [148] BiocParallel_1.44.0     mclust_6.1.2            GOSemSim_2.36.0        </span></span>
<span><span class="co">#&gt; [151] R.oo_1.27.1             magrittr_2.0.4          modeltools_0.2-24      </span></span>
<span><span class="co">#&gt; [154] ggplotify_0.1.3         wordcloud_2.6           dotCall64_1.2          </span></span>
<span><span class="co">#&gt; [157] patchwork_1.3.2         Rcpp_1.1.1              ape_5.8-1              </span></span>
<span><span class="co">#&gt; [160] ggnewscale_0.5.2        gdtools_0.5.0           reticulate_1.44.1      </span></span>
<span><span class="co">#&gt; [163] stringi_1.8.7           MASS_7.3-65             org.Hs.eg.db_3.22.0    </span></span>
<span><span class="co">#&gt; [166] plyr_1.8.9              flexmix_2.3-20          ggstats_0.12.0         </span></span>
<span><span class="co">#&gt; [169] parallel_4.5.2          listenv_0.10.0          ggrepel_0.9.6          </span></span>
<span><span class="co">#&gt; [172] deldir_2.0-4            Biostrings_2.78.0       splines_4.5.2          </span></span>
<span><span class="co">#&gt; [175] tensor_1.5.1            hms_1.1.4               igraph_2.2.1           </span></span>
<span><span class="co">#&gt; [178] spatstat.geom_3.7-0     RcppHNSW_0.6.0          reshape2_1.4.5         </span></span>
<span><span class="co">#&gt; [181] stats4_4.5.2            evaluate_1.0.5          SeuratObject_5.3.0     </span></span>
<span><span class="co">#&gt; [184] tzdb_0.5.0              tweenr_2.0.3            httpuv_1.6.16          </span></span>
<span><span class="co">#&gt; [187] openssl_2.3.4           RANN_2.6.2              tidyr_1.3.2            </span></span>
<span><span class="co">#&gt; [190] purrr_1.2.1             polyclip_1.10-7         future_1.69.0          </span></span>
<span><span class="co">#&gt; [193] clue_0.3-66             scattermore_1.2         ggplot2_4.0.2          </span></span>
<span><span class="co">#&gt; [196] gridBase_0.4-7          ggforce_0.5.0           xtable_1.8-4           </span></span>
<span><span class="co">#&gt; [199] RSpectra_0.16-2         tidytree_0.4.7          tidydr_0.0.6           </span></span>
<span><span class="co">#&gt; [202] later_1.4.5             viridisLite_0.4.3       class_7.3-23           </span></span>
<span><span class="co">#&gt; [205] ragg_1.5.0              tibble_3.3.1            clusterProfiler_4.18.4 </span></span>
<span><span class="co">#&gt; [208] aplot_0.2.9             memoise_2.0.1           AnnotationDbi_1.72.0   </span></span>
<span><span class="co">#&gt; [211] IRanges_2.44.0          cluster_2.1.8.1         globals_0.19.0         </span></span>
<span><span class="co">#&gt; [214] concaveman_1.2.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Konstantinos Antonopoulos, Maria Bueno Alvez, Emilio Skarwan, Mathias Uhlen.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer>
</div>





  </body>
</html>
